lib/semantics/mod-file.cc

```diff
@@ -590,53 +608,65 @@ std::ostream &PutLower(std::ostream &os, const std::string &str) {
   return os;
 }
 
-// Write the module file at path, prepending header. Return false on error.
-static bool WriteFile(const std::string &path, std::string &&contents) {
-  std::fstream stream;
-  auto header{GetHeader(contents)};
-  auto size{GetFileSize(path)};
-  if (size == header.size() + 1 + contents.size()) {
-    // file exists and has the right size, check the contents
-    stream.open(path, std::ios::in | std::ios::out);
-    if (FileContentsMatch(stream, header, contents)) {
-      return true;
+// Write the module file at path, prepending header. If an error occurs,
+// return an error message, otherwise nullptr.
+static const char *WriteFile(
+    const std::string &path, const std::string &contents) {
+  std::FILE *fp{std::fopen(path.c_str(), "a+")};
+  if (!fp) {
+    return std::strerror(errno);
+  }
+  if (flock(fileno(fp), LOCK_EX | LOCK_NB) != 0) {
+    fclose(fp);
+    return "Another compilation is currently using it";
+  }
+  std::size_t size = std::ftell(fp);
```

Comment:
Are you using `=` to get a cast?  It's more clear to use `static_cast<>`.

Reply:
OK